django/channels/routing.py

```diff
@@ -0,0 +1,215 @@
+from __future__ import unicode_literals
+
+import importlib
+import re
+
+from django.core.exceptions import ImproperlyConfigured
+from django.utils import six
+
+from .utils import name_that_thing
+
+
+class Router(object):
+    """
+    Manages the available consumers in the project and which channels they
+    listen to.
+
+    Generally this is attached to a backend instance as ".router"
+    """
+
+    def __init__(self, routing):
+        # Use a blank include as the root item
+        self.root = Include(routing)
+        # Cache channel names
+        self.channels = self.root.channel_names()
+
+    def add_route(self, route):
+        """
+        Adds a single raw Route to us at the end of the resolution list.
+        """
+        self.root.routing.append(route)
+        self.channels = self.root.channel_names()
+
+    def match(self, message):
+        """
+        Runs through our routing and tries to find a consumer that matches
+        the message/channel. Returns (consumer, extra_kwargs) if it does,
+        and None if it doesn't.
+        """
+        # TODO: Maybe we can add some kind of caching in here if we can hash
+        # the message with only matchable keys faster than the search?
+        return self.root.match(message)
+
+    def check_default(self, http_consumer=None):
+        """
+        Adds default handlers for Django's default handling of channels.
+        """
+        # We just add the default Django route to the bottom; if the user
+        # has defined another http.request handler, it'll get hit first and run.
+        # Inner import here to avoid circular import; this function only gets
+        # called once, thankfully.
+        from django.core.handlers.asgi import ViewConsumer
+        self.add_route(Route("http.request", http_consumer or ViewConsumer()))
+
+    @classmethod
+    def resolve_routing(cls, routing):
+        """
+        Takes a routing - if it's a string, it imports it, and if it's a
+        dict, converts it to a list of route()s. Used by this class and Include.
+        """
+        # If the routing was a string, import it
+        if isinstance(routing, six.string_types):
+            module_name, variable_name = routing.rsplit(".", 1)
```

Comment:
Another option would be using `import_string`: https://docs.djangoproject.com/en/1.9/ref/utils/#module-django.utils.module_loading